R/correctionsataglance-data.R

```diff
@@ -342,13 +344,15 @@ findTextLocations <- function(dataIn, isDateData = FALSE, ...){
   return(list(x = x, y = y))
 }
 
-isTextLong <- function(labelText, dateLim, startD, endD){
-  early <- which(startD < dateLim[1])
-  late <- which(endD > dateLim[2])
-  startD[early] <- dateLim[1]
-  endD[late] <- dateLim[2]
+isTextLong <- function(labelText, dateLim = NULL, startD, endD, totalDays = NULL){
+  if(is.null(totalDays)){
+    early <- which(startD < dateLim[1])
+    late <- which(endD > dateLim[2])
+    startD[early] <- dateLim[1]
+    endD[late] <- dateLim[2]
+    totalDays <- difftime(dateLim[2], dateLim[1], units="days")
+  } 
   
-  totalDays <- difftime(dateLim[2], dateLim[1], units="days")
   widthOfChar <- (1/365)*totalDays*3 #each character will be 1/365 * num days in the range * buffer
   widthOfLabel <- nchar(labelText)*widthOfChar
   widthOfRect <- difftime(endD, startD, units="days")
```

Comment:
the way I'm passing parameters in, for example with example4.json, I don't want Apr 2016 to plot because the rectangle for that month is too small at the end of the sequence. The values coming in are 

labelText = Apr 2016
dateLim = NULL
startD = 2015-04-01
endD = 2016-04-11
totalDays = 376

widthOfChar for the labelText = 3.0904...
widthofLabel = 24.7232
widthOfRect = 376
so, when widthOfRect is compared to widthOfLabel, even on the last one, it fits, because it looks like it's comparing to the full length rather than the short rect available... I think.